app/code/Magento/CheckoutAgreements/Model/ResourceModel/Agreement/Grid/Collection.php

```diff
@@ -63,7 +63,7 @@ private function getStoresForAgreements()
 
         if (!empty($agreementId)) {
             $select = $this->getConnection()->select()->from(
-                ['agreement_store' => 'checkout_agreement_store']
+                ['agreement_store' => $this->getResource()->getTable('checkout_agreement_store')]
```

Comment:
Is `getResource()` really necessary?

Reply:
@gelanivishal Just `$this->getTable(...)` should be enough to get prefix table name.